target.c

```diff
@@ -329,8 +330,13 @@ int tcmu_add_dev_to_recovery_list(struct tcmu_device *dev)
 		ret = -ENOMEM;
 		goto done;
 	}
+
+	pthread_attr_init(&attr);
+	pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED);
+
 	ret = pthread_create(&tpg->recovery_thread, NULL,
```

Comment:
Since the `tpg->recovery_thread` member is not used in any other place after, we can get rid of it by using a local tmp parameter here.